### ISSUE TITLE
Cannot detect lightning testnet service in system

### ISSUE THREAD

**Issue by zerotobtc** [ElementsProject/lightning#5009]
### Issue and Steps to Reproduce

I installed lightning on my testnet node according to the [installation guide](https://github.com/ElementsProject/lightning/blob/master/doc/INSTALL.md#to-build-on-ubuntu). However, when I ran `lightningd` I got the following error:

```
INFO    plugin-bcli: bitcoin-cli initialized and connected to bitcoind.
Wrong network! Our Bitcoin backend is running on 'test', but we expect 'main'.
```

When I do `lightningd --testnet` it starts just fine.

When I run `systemctl start lightningd` and after that `systemctl status lightningd` I see that it has failed to start:

```
● lightningd.service - c-lightning daemon
     Loaded: loaded (/etc/systemd/system/lightningd.service; disabled; vendor pres>
     Active: failed (Result: exit-code) since XXXXXXXXXXXXXXX 3s ago
    Process: 247313 ExecStart=/usr/local/bin/lightningd --testnet (code=exited, st>
   Main PID: 247313 (code=exited, status=1/FAILURE)
        CPU: 123ms

XXXXXXXXXX btc-testnet systemd[1]: lightningd.service: Scheduled restart job,>
XXXXXXXXXX btc-testnet systemd[1]: Stopped c-lightning daemon.
XXXXXXXXXX btc-testnet systemd[1]: lightningd.service: Start request repeated>
XXXXXXXXXX btc-testnet systemd[1]: lightningd.service: Failed with result 'ex>
XXXXXXXXXX btc-testnet systemd[1]: Failed to start c-lightning daemon.
```

What is strange is that I can somehow use lightning-cli commands such as `lnc --testnet getinfo` (and it works fine), but for example running it without the `--testnet` flag gives me `lightning-cli: Connecting to 'lightning-rpc': No such file or directory`... 

I can run all commands with the `--testnet` flag. So there is some lightning service already running...

What I also tried is creating a `lightningd.service` as they did in [this guide](https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/blob/master/19_1_Verifying_Your_Lightning_Setup.md#run-lightningd-as-a-service) but I get the same 'failed' error message.

Basically, I just want to know how to SEE that the testnet lightning service is running. And what highlights a testnet installation of c-lightning? It seems to me that you kind of always install a mainnet installation because you always need to add the `--testnet` flag to the commands.

Thanks in advance for any help, explanations and reading sources!

**Comment by vincenzopalazzo** [comment]
It is a simple conf problem, historically c-lighting had a single conf file in the root of the `.lightning` where you need to specify the `network={bitcoin|testnet|litecoin|liquid}`.

To avoid specifying the network in the conf file, it is possible to divide the conf file in a subdirectory, like `.lightnong/bitcoin`, `.lightnong/testnet`, etc

Where you must not provide the network flag in the conf.

So, to solve your problem and run your `lightningd` with the `lightningd.service` you should provide on the root of the directory a `.lightning/conf `file where there is at least the `network=testnet` flag.

What I personally do, is run a script for each restart of the system and add a` --testnet` flag, and provide the conf file in the subdirectory without network propriety

Maybe there is another trick to do with `lightningd.service`

**Comment by cdecker** [comment]
Sure, pass the `--network=testnet` argument in the systemd unit file, and it'll use that. This also allows you to run multiple networks without having a network specific  `config` in the lightning dir itself.

**Comment by zerotobtc** [comment]
@cdecker in my unit file it already says

```
[Service]
ExecStart=/usr/local/bin/lightningd --testnet
```
That's what you meant, isn't it?

**Comment by zerotobtc** [comment]
@vincenzopalazzo Thanks a lot, now the service is running!

**Comment by cdecker** [comment]
> @cdecker in my unit file it already says
> 
> ```
> [Service]
> ExecStart=/usr/local/bin/lightningd --testnet
> ```
> 
> That's what you meant, isn't it?

Correct, that's how I configure mine. Glad it worked 🙂

**Comment by Geoelycom** [comment]
> @cdecker in my unit file it already says
> 
> ```
> [Service]
> ExecStart=/usr/local/bin/lightningd --testnet
> ```
> 
> That's what you meant, isn't i

> @cdecker in my unit file it already says
> 
> ```
> [Service]
> ExecStart=/usr/local/bin/lightningd --testnet
> ```
> 
> That's what you meant, isn't it?

I'm having the same issue right now. can you please tell me where I can locate this unit file? or do I just create a config file in the root of lightning as @vincenzopalazzo suggested and set network=testnet? please someone help me out here

**Comment by Geoelycom** [comment]
> @cdecker in my unit file it already says
> 
> ```
> [Service]
> ExecStart=/usr/local/bin/lightningd --testnet
> ```
> 
> That's what you meant, isn't it?

can you please tell me where I can find this file please? I'm having the same issue
